### PR TITLE
[FIX] Avoid silent divergence by reporting inf losses

### LINKED ISSUE #1601: [Losses] _divide_no_nan turns an infinite loss into 0.0, so broken trials win Auto search
### What happened + What you expected to happen

What happened:
A diverged trial reported loss = 0.0. Its predictions were ±inf, so the loss
should have been inf. Because 0.0 is the lowest possible value, the Auto
search treated the broken trial as the best one, kept it through every ASHA
rung, and returned it as the winner. No warning, no exception. best_config
stored the diverged configuration.

What I expected to happen:
A diverged trial should report inf, or at least a very large loss, so it
ranks worst in the search. The winner should be one of the trials that
actually converged.

### Versions / Dependencies

```
neuralforecast 3.2.1
torch 2.13.0+cu130
ray 2.57.0
python 3.12.3
Linux
```

### Reproduction script

```python
import torch
from neuralforecast.losses.pytorch import MAE

y = torch.tensor([[[10.0], [20.0]]])
print(MAE()(y=y, y_hat=torch.full_like(y, float("inf"))))

# actual:   tensor(0.)
# expected: tensor(inf)
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by marcopeix**
Fixes #1601.  

A diverged trial reported loss `0.0`, the lowest possible value, so it won every Auto search.

`_divide_no_nan` sanitized the quotient instead of guarding the denominator. Every `inf`/`nan` loss reduced to `0.0`.

## Changes

- `_divide_no_nan`: guard `b == 0` instead of scrubbing the result. Non-finite losses propagate; the deliberate `y == 0` skips in MAPE/SMAPE/MASE.
- `_weighted_mean`: drop zero-weight entries first, so a masked-out `inf` doesn't become `nan * 0`.
- `weighted_average`: `clamp(min=1.0)` under-normalized when total weight was below 1, and `if dim` treated `dim=0` as falsy.
- Validation reports a non-finite `ptl/val_loss` as `+inf` with a warning, so a diverged trial ranks worst instead of erroring the search.
Training keeps its NaN raise; `inf` only warns (mixed precision recovers).

### CODE REVIEW

**Review by nasaul — APPROVED** [review]
LGTM
